src/components/input-time/input-time.tsx

```diff
@@ -0,0 +1,872 @@
+   */
+  @Event() calciteInputTimeInput: EventEmitter<string>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteInternalInputTimeBlur: EventEmitter<void>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteInternalInputTimeFocus: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("blur")
+  blurHandler(): void {
+    this.activeEl = undefined;
+    this.calciteInternalInputTimeBlur.emit();
+    this.emitChangeIfUserModified();
+  }
+
+  @Listen("click")
+  clickHandler(event: MouseEvent): void {
+    const composedEventPath = event.composedPath();
+    if (composedEventPath.includes(this.hourEl)) {
+      return;
+    }
+    if (composedEventPath.includes(this.minuteEl)) {
+      return;
+    }
+    if (composedEventPath.includes(this.secondEl)) {
+      return;
+    }
+    if (composedEventPath.includes(this.meridiemEl)) {
+      return;
+    }
+    this.setFocus();
+  }
+
+  @Listen("focus")
+  focusHandler(): void {
+    this.calciteInternalInputTimeFocus.emit();
+  }
+
+  @Listen("keydown")
+  keyDownHandler({ defaultPrevented, key }: KeyboardEvent): void {
+    if (key === "Enter" && !defaultPrevented && !this.disabled) {
+      this.emitChangeIfUserModified();
+      submitForm(this);
+      return;
+    }
+
+    switch (this.activeEl) {
+      case this.hourEl:
+        if (key === "ArrowRight") {
+          this.setFocus("minute");
+        }
+        break;
+      case this.minuteEl:
+        switch (key) {
+          case "ArrowLeft":
+            this.setFocus("hour");
+            break;
+          case "ArrowRight":
+            if (this.step !== 60) {
+              this.setFocus("second");
+            } else if (this.hourCycle === "12") {
+              this.setFocus("meridiem");
+            }
+            break;
+        }
+        break;
+      case this.secondEl:
+        switch (key) {
+          case "ArrowLeft":
+            this.setFocus("minute");
+            break;
+          case "ArrowRight":
+            if (this.hourCycle === "12") {
+              this.setFocus("meridiem");
+            }
+            break;
+        }
+        break;
+      case this.meridiemEl:
+        switch (key) {
+          case "ArrowLeft":
+            if (this.step !== 60) {
+              this.setFocus("second");
+            } else {
+              this.setFocus("minute");
+            }
+            break;
+        }
+        break;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * Sets focus on the component.
+   *
+   * @param target
+   */
+  @Method()
+  async setFocus(target?: TimePart): Promise<void> {
+    this[`${target || "hour"}El`]?.focus();
```

Comment:
I forget the name, but you need to await a util from `loadable.ts` before calling `focus()`. See any other components for an example.